src/Components/Components/src/ComponentBase.cs

```diff
@@ -224,19 +233,19 @@ private Task CallOnParametersSetAsync()
             StateHasChanged();
 
             return shouldAwaitTask ?
-                ProcessLifecycleTaskAsync(task) :
+                CallStateHasChangedOnAsyncCompletion(task) :
                 Task.CompletedTask;
         }
 
-        private async Task ProcessLifecycleTaskAsync(Task task)
+        private async Task CallStateHasChangedOnAsyncCompletion(Task task)
         {
             try
             {
                 await task;
             }
             catch when (task.IsCanceled)
```

Comment:
Nit: Is there something specific you can catch in here instead of using task.IsCancelled? It's just weird to read a catch block with a when clause instead of an exception. It's not immediate how the exception thrown and the condition are related.

Reply:
We get a mix of two different exceptions depending on how the Task was cancelled. Throwing a `OperationCanceledException` (`cancelationToken.ThrowIfCancellationRequested()`) does this. The `Task` becomes cancelled, but the exception continues to be `OperationCancelledException`. Alternatively awaiting a cancelled task `Task.FromCanceled` results in a `TaskCancellationException` which also has the same state. Observing the state of the task seems like less work \ easier to grock.

Reply:
See https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/task-cancellation. The thing is a bit involved.

Reply:
Can you add a comment? I'm happy if we have a comment so that people can understand it.

Reply:
Done